packages/firebase_ml/test/firebase_ml_test.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:firebase_ml/firebase_ml.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  String MODEL_NAME = "myModelName";
+  String MODEL_HASH = "myModelHash";
+  String MODEL_FILE_PATH = "someDestination";
+
+  group('$FirebaseModelManager', () {
+    final List<MethodCall> log = <MethodCall>[];
+    final FirebaseModelManager modelManager = FirebaseModelManager.instance;
+
+    setUp(() {
+      FirebaseModelManager.channel
+          .setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+
+        switch (methodCall.method) {
+          case 'FirebaseModelManager#download':
+            var model = <String, String>{};
+            model['modelName'] = MODEL_NAME;
+            model['modelHash'] = MODEL_HASH;
+            return model;
+          case 'FirebaseModelManager#getLatestModelFile':
+            return MODEL_FILE_PATH;
+          case 'FirebaseModelManager#isModelDownloaded':
+            return true;
+          default:
+            return null;
+        }
+      });
+      log.clear();
+    });
+
+    tearDown(() {
+      FirebaseModelManager.channel.setMockMethodCallHandler(null);
+    });
+
+    test('valid sequence of calls', () async {
+      expect(modelManager, isNotNull);
+      FirebaseCustomRemoteModel model = FirebaseCustomRemoteModel(MODEL_NAME);
+      expect(model.modelHash, isNull);
+      FirebaseModelDownloadConditions conditions =
+          FirebaseModelDownloadConditions(requireWifi: true);
+      expect(conditions, isNotNull);
```

Comment:
not really related to the flow of downloading a model, it's more of a test for making sure these classes work as intended, maybe pull them into separate tests?

Reply:
saw two examples of tests in different plugins, one doing altogether, another separately. This was you know which statement fails (because you follow expects along the way). I have no strong preference for either way, can change.

Reply:
Done

Reply:
If you break them down into smaller tests it's even easier to see what fails because you'll see it in the test description without even having to follow the expects

Reply:
Done